source/docs/software/commandbased/commands-v3/binding-commands-to-triggers-v3.rst

```diff
@@ -0,0 +1,425 @@
+
+## Binding Methods
+
+### ``onTrue(Command)``
+
+Run the command when the trigger goes from false to true (rising edge).
+
+.. code-block:: java
+
+   driver.a().onTrue(
+     intake.run(coro -> intake.extend()).named("Extend Intake")
+   );
+
+The command runs once per button press, even if you hold the button.
+
+### ``onFalse(Command)``
+
+Run the command when the trigger goes from true to false (falling edge).
+
+.. code-block:: java
+
+   driver.a().onFalse(
+     intake.run(coro -> intake.retract()).named("Retract Intake")
+   );
+
+### ``whileTrue(Command)``
+
+Run the command while the trigger is true. Cancel when it becomes false.
+
+.. code-block:: java
+
+   driver.rightBumper().whileTrue(
+     intake.run(coro -> {
+       intake.setSpeed(0.8);
+       coro.park(); // Run until canceled
+     })
+     .whenCanceled(() -> intake.setSpeed(0))
+     .named("Run Intake")
+   );
+
+This is perfect for "hold button to do action" patterns.
+
+### ``whileFalse(Command)``
+
+Run the command while the trigger is false. Cancel when it becomes true.
+
+.. code-block:: java
+
+   new Trigger(() -> sensor.beamBroken())
+     .whileFalse(
+       intake.runRepeatedly(() -> intake.run()).named("Run Until Beam Break")
+     );
+
+### ``toggleOnTrue(Command)``
+
+Toggle the command on and off with each button press.
+
+.. code-block:: java
+
+   driver.x().toggleOnTrue(
+     intake.run(coro -> {
+       intake.setSpeed(0.8);
+       coro.park();
+     })
+     .whenCanceled(() -> intake.setSpeed(0))
+     .named("Toggle Intake")
+   );
+
+First press: start command. Second press: cancel command. Third press: start again.
+
+## Common Patterns
+
+### Pattern 1: Instant Action on Press
+
+.. code-block:: java
+
+   driver.a().onTrue(
+     pneumatics.run(coro -> solenoid.toggle()).named("Toggle Solenoid")
+   );
+
+### Pattern 2: Hold to Run
+
+.. code-block:: java
+
+   driver.rightBumper().whileTrue(
+     intake.run(coro -> {
+       intake.on();
+       coro.park();
+     })
+     .whenCanceled(() -> intake.off())
+     .named("Hold Intake")
+   );
+
+### Pattern 3: Toggle
+
+.. code-block:: java
+
+   driver.x().toggleOnTrue(
+     climber.run(coro -> {
+       climber.extend();
+       coro.park();
+     })
+     .whenCanceled(() -> climber.retract())
+     .named("Toggle Climber")
+   );
+
+### Pattern 4: Two Buttons for Opposite Actions
+
+.. code-block:: java
+
+   driver.leftBumper().whileTrue(
+     shooter.run(coro -> {
+       shooter.spinUp();
+       coro.park();
+     })
+     .whenCanceled(() -> shooter.stop())
+     .named("Spin Up")
+   );
+
+   driver.leftTrigger().whileTrue(
+     shooter.run(coro -> {
+       shooter.spinReverse();
+       coro.park();
+     })
+     .whenCanceled(() -> shooter.stop())
+     .named("Spin Reverse")
+   );
+
+### Pattern 5: Conditional Binding
+
+.. code-block:: java
+
+   Trigger canShoot = new Trigger(() -> shooter.atSpeed())
+     .and(new Trigger(() -> turret.onTarget()))
```

Comment:
`and` accepts a `BooleanSupplier`; there's no need to wrap it in a trigger (also applies in a few other examples on this page)


```suggestion
     .and(turret::onTarget)
```